### PR TITLE
Do not display legacy NVIDIA driver if current is available

### PR DESCRIPTION
**Pull request by mmstick**
Even if the legacy driver is supported, it's confusing to ask the user to install an older driver than their hardware supports. So, this will do a check if the 340 or 390 driver appears in the list from `ubuntu-drivers`, and ignores them if a non-legacy driver is found.

### CODE REVIEW

File: src/Core/Client.vala [hunk 1]
```diff
@@ -794,3 +800,33 @@ public class AppCenterCore.Client : Object {
...
+        } else if (driver.has_prefix("nvidia-driver")) {
+            flags |= DriverFlags.NVIDIA_CURRENT;
+        }
+    }
+
+    return flags;
+}
+
+private bool driver_is_nvidia_legacy (string driver) {
+    return driver == "nvidia-driver-390" || driver == "nvidia-driver-340";
```
Comment (brs17) [inline src/Core/Client.vala]: Do we want to hardcode these legacy versions in or just take the older of two listed versions as the legacy version? Would we ever see anything else listed from `ubuntu-drivers list`?

Comment (mmstick) [reply src/Core/Client.vala]: We could simply ignore everything but the highest version available.

**Comment by cassidyjames** [comment]
I _believe_ we've heard of issues on older generations of cards where older drivers are actually necessary or more performant for some reason, which is why Ubuntu offers them up. I'm not sure how true that still is, though.

**Comment by mmstick** [comment]
There's been a lot of confusion lately about the shop listing 390 driver as installable, even though it also lists that they have the 435 driver installed, so they try to install the 390 driver, and end up breaking the system in the process.

File: src/Core/Client.vala (hunk 1 above)
Comment (mmstick) [reply src/Core/Client.vala]: Okay, the branch has been revised to only display the latest NVIDIA driver supported.

**Pull request merged**